lib/cpp/src/thrift/transport/TSSLServerSocket.cpp

```diff
@@ -48,7 +48,12 @@ TSSLServerSocket::TSSLServerSocket(int port,
 }
 
 boost::shared_ptr<TSocket> TSSLServerSocket::createSocket(THRIFT_SOCKET client) {
-  return factory_->createSocket(client);
+  boost::shared_ptr<THRIFT_SOCKET> interruptSocketReader = getInterruptSocket();
```

Comment:
The way I implemented this in TSocket was to provide two constructors - the second one takes the interrupt socket as the second argument.  If the same logic is applied here, there would be no need to expose it via the method added to TServerSocket called getInterruptSocket.